### DISCUSSION TITLE
Something like the continue keyword, but for recursion

### DISCUSSION THREAD

**Discussion by Pjotor87** [dotnet/csharplang#5471] [Language Ideas]
The continue keyword that can be used in loops to start a new iteration - I wonder if it is possible to either be able to use continue outside of a loop or have a smiliar keyword available like "recurse" to recurse a function using the same parameters as is currently being used.

Usually when I see recursion used there's usually always a code comment that goes along with the recursion call like 
// recurse.
Just to quickly make the next person aware and not have to think to figure that out. Also, making recursive calls in functions with a lot of parameters can be hard to both write and read.

Have a great day!

/Peter

**Comment by 333fred** [comment]
Generally speaking, if I see a recursive call with the same parameters, I know that I've found an infinite loop that will stack overflow. Given that, I'm not sure that this is a good idea.

**Reply by TahirAhmadov** [reply]
I think he meant the parameter variables are modified somehow, so the recursion is meaningfully limited. Obviously calling the method with the same parameters even without the keyword would produce a stack overflow.

**Comment by svick** [comment]
I am also quite confused about the need to recurse into the same function with the same parameters. Could you give an example where that would be useful? Would making the recursive function into a local function that captures the parameters help?

**Comment by Xyncgas** [comment]
maybe if we can catch stack overflow through some hack we solve turing's halting problem

**Comment by Rekkonnect** [comment]
I'm betting that you can simply modify your current arguments before `recurse`, which keyword would replace manually recursively calling the function. An (obviously impractical) example:

```csharp
public void Fill<T>(T[] array, T value, int index)
{
    if (index >= array.Length)
        return;
    array[index] = value;
    index++;
    recurse;
}
```
